lib/plugins/aws/package/compile/events/apiGateway/lib/restApi.js

```diff
@@ -54,6 +54,26 @@ module.exports = {
           });
     }
 
+    if (!_.isEmpty(this.serverless.service.provider.apiGateway) &&
+      !_.isEmpty(this.serverless.service.provider.apiGateway.apiKeySourceType)) {
+      const apiKeySourceType =
+        this.serverless.service.provider.apiGateway.apiKeySourceType.toUpperCase();
+      const validApiKeySourceType = ['HEADER', 'AUTHORIZER'];
+
+      if (!_.includes(validApiKeySourceType, apiKeySourceType)) {
+        const message =
+          'apiKeySourceType must be one of "HEADER" or "AUTHORIZER". ' +
+          `You provided ${apiKeySourceType}.`;
+        throw new this.serverless.classes.Error(message);
```

Comment:
Use async code as the following since we should not mix sync and async ones as possible as we can.

```
return BbPromise.reject(new this.serverless.classes.Error(message));
```

Reply:
I see! I will revise it according to your advice, thank you.

and I want to ask a question, there are same error handling code in this file like [this](https://github.com/serverless/serverless/blob/92e3f0b5987a05b1eb78658949874b7a55947a5f/lib/plugins/aws/package/compile/events/apiGateway/lib/restApi.js#L22) and [this](https://github.com/serverless/serverless/blob/92e3f0b5987a05b1eb78658949874b7a55947a5f/lib/plugins/aws/package/compile/events/apiGateway/lib/restApi.js#L29).
Should I fix those code as well?

Reply:
@aki-webii 
>Should I fix those code as well?

Yup, Would be great if you can fix it as well 👍